lib/ArduinoNative/src/WiFiUdp.cpp

```diff
@@ -62,30 +71,67 @@
 
 uint8_t WiFiUDP::begin(uint16_t port)
 {
-    UTIL_NOT_USED(port);
+    m_socket = socket(AF_INET, SOCK_DGRAM, 0);
+
+    // setup udp sockt with 10 us timout, would block calling function indefinitely
+    struct timeval read_timeout;
+    read_timeout.tv_sec  = 0;
+    read_timeout.tv_usec = 10;
+    setsockopt(m_socket, SOL_SOCKET, SO_RCVTIMEO, &read_timeout, sizeof(read_timeout));
+
+    if (m_socket == -1)
+    {
+        LOG_ERROR("socket creation failed");
+    }
+    else
+    {
+        m_servaddr.sin_family = AF_INET;
+        return 1;
+    }
+
     return 0;
 }
 
 void WiFiUDP::stop()
 {
+    close(m_socket);
 }
 
 int WiFiUDP::beginPacket(IPAddress ip, uint16_t port) // NOLINT(performance-unnecessary-value-param)
 {
-    UTIL_NOT_USED(ip);
-    UTIL_NOT_USED(port);
-    return 0;
+    LOG_DEBUG("%s %d", ip.toString().c_str(), port);
+    m_servaddr.sin_port        = htons(port);
+    m_servaddr.sin_addr.s_addr = ip.raw();
+    m_outBufferLength          = 0;
+
+    const char* hello = "Hello from client";
+
+    sendto(m_socket, (const char*)hello, strlen(hello), 0, (const struct sockaddr*)&m_servaddr, sizeof(m_servaddr));
+    return 1;
 }
 
 size_t WiFiUDP::write(const uint8_t* buffer, size_t length)
 {
-    UTIL_NOT_USED(buffer);
-    UTIL_NOT_USED(length);
-    return 0U;
+    memcpy(m_outBuffer, buffer, length);
```

Comment:
missing range check

Reply:
Added range check. Bytes exceeding the internal buffer length will be dropped for now. This will be changed with future data structure and compatibility improvements.